erpnext/shopping_cart/cart.py

```diff
@@ -276,20 +276,24 @@ def set_taxes(quotation, cart_settings):
 # 	# append taxes
 	quotation.append_taxes_from_master()
 
-def get_customer(user=None):
+def get_party(user=None):
 	if not user:
 		user = frappe.session.user
 
-	customer = frappe.db.get_value("Contact", {"email_id": user}, "customer")
+	party = frappe.db.get_value("Contact", {"email_id": user}, ["customer", "supplier"], as_dict=1)
+
+	if party:
+		party, doctype = get_party_detail(party)
+
```

Comment:
This could be simply written as:

```
party = frappe.db.get_value("Contact", {"email_id": user}, ["customer", "supplier"], as_dict=1)
if party:
    party_doctype = 'Customer' if party.customer else 'Supplier'
    party = party.customer or party.supplier

...

    if party:
        return frappe.get_doc(party_doctype, party)

    else:
        ...
```